scene/resources/mesh.cpp

```diff
@@ -808,8 +807,9 @@ void ArrayMesh::add_surface_from_arrays(PrimitiveType p_primitive, const Array &
 				aabb.expand_to(vtx[i]);
 		}
 
-		surfaces.write[surfaces.size() - 1].aabb = aabb;
-		surfaces.write[surfaces.size() - 1].is_2d = arr.get_type() == Variant::POOL_VECTOR2_ARRAY;
+		s.aabb = aabb;
+		s.is_2d = arr.get_type() == Variant::POOL_VECTOR2_ARRAY;
+		surfaces.push_back(s);
```

Comment:
There's one potential issue here, as there might be an early return on line 796 with `ERR_FAIL_COND(len == 0);`, so then `surfaces` would no longer get this extra default-initialized `Surface` appended.
I can't assess if that would be a problem and/or the previous code was intended behaviour though, just raising the concern.

Reply:
You are right. In this case I would prefer to not to add to the vector when the assert is triggered. I guess no value is better than an uninitialized value?

Reply:
I tend to agree. I just wondered because we still have the call to `VS::mesh_add_surface_from_arrays` initially, so I just want to check that it doesn't imply that the `surfaces` vector should be somewhat in sync with whatever the VS allocates.

I'm not asking you to check that though, I'm just commenting here to then ask @reduz to confirm that it's fine as is, or if it should be done differently.

Reply:
@reduz says it seems fine, so let's merge :)